### PR TITLE
 #464 #461 Move slider to its own file and fix related bugs

### PR DESCRIPTION
**Pull request by MelindaSW**
- Moved the slider to its own file, named it CustomScrollBar and added some quick styling before positioning it by the logviewer. 

- I also renamed the slider code in the LogViewer to have names related to scroll instead of slider.   

-  Created a log viewer root container and placed the custom scroller next to the log viewer. It now stays where it should even when the lines are long. 

-  Added overflow-x: auto to the logviewer container to make the length of the file readable.

-  Finally some issues occurred when calculating the amount of lines to fill the viewer. It was based on offsetHeight. It did not take the horisontal scrollbar into account. I changed offsetHeight to clientHeight and it seems to be more accurate.

### CODE REVIEW

File: src/js/view/components/LogViewerList.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ const LogViewerList = props => {
   const oneCharacterSizeRef = useRef();
   const [characterDimensions, setCharacterDimensions] = useState({
     width: 10,
-    height: 17
+    height: 19
```
Comment (Doverstav) [inline src/js/view/components/LogViewerList.js]: Has the height of a character changed? Might be beacuse we switched fonts when moving to Fabric UI I guess?

File: src/js/view/components/LogViewerList.js [hunk 2]
```diff
@@ -107,11 +107,11 @@ const LogViewerList = props => {
   }, [props.lines]);
 
   useEffect(() => {
-    // Calculating the correct amount of lines needed to fill the page in the logviewer
+    // Calculating the amount of lines needed to fill the page in the logviewer
     setNumberOfLinesToFillLogView(
       Math.round(props.containerHeight / characterDimensions.height)
     );
-  }, [props.containerHeight]);
+  }, [props.containerHeight, characterDimensions]);
```
Comment (Doverstav) [inline src/js/view/components/LogViewerList.js]: Why is this needed? Do we expect that character dimensions to change?

File: src/js/view/components/LogViewerList.js (hunk 1 above)
Comment (MelindaSW) [reply src/js/view/components/LogViewerList.js]: I had it changed from 19 to 17 in an earlier commit since I thought it made a difference then. I tried changing it back to solve things now but did not notice any difference. Looking at the program on the laptop and on the bigger screen has caused some issues with the line numbers for me from time to time. Thought I'd leave it in the state it was before I started messing around with it for now.

Comment (MelindaSW) [reply src/js/view/components/LogViewerList.js]: I did not try the trackpad. Should have thought of that. Good thing you noticed.

File: src/js/view/components/LogViewerList.js (hunk 2 above)
Comment (MelindaSW) [reply src/js/view/components/LogViewerList.js]: The character dimensions update on resize as seen in this effect. I found it useful to trigger the recalculation of the amount of lines in the viewer then. But I could be wrong.
```
useEffect(() => {
    // Handler to update the dimensions when needed
    const handleResize = () => {
      setListDimensions({
        width: logViewerListContainerRef.current.offsetWidth,
        height: logViewerListContainerRef.current.offsetHeight
      });

      setCharacterDimensions({
        width: oneCharacterSizeRef.current.offsetWidth,
        height: oneCharacterSizeRef.current.offsetHeight
      });
    };
    handleResize();

    // Calls are throttled to once every 200 ms
    const debouncedResizeHandler = _.debounce(handleResize, 200);
    window.addEventListener('resize', debouncedResizeHandler);
    // Return cleanup function
    return () => {
      window.removeEventListener('resize', debouncedResizeHandler);
    };
  }, []);
```

Comment (Doverstav) [reply src/js/view/components/LogViewerList.js]: Ah, of course. I thought they never updated. Makes sense to recalculate lines to fill then, nice catch 👍
